py-rse/style.Rmd

```diff
@@ -338,65 +386,115 @@ if __name__ == '__main__':
+    onto specific lines
+    (something that GitHub unfortunately doesn't yet support).
 
-or:
+Ask for clarification.
+:   If you don't understand something,
+    or don't understand why the author did it,
+    ask.
+    (And when the author explains it,
+    think about suggesting that the explanation ought to be documented somewhere.)
 
-1.  Figure out what the user has asked for.
-2.  Repeatedly:
-    1.  Wait for user input.
-    2.  Do what the user has asked.
-3.  Exit when a "stop" command of some sort is received.
+Offer alternatives.
+:   Telling authors that something is wrong is helpful;
+    telling them what they might do instead is more so.
 
-Each step in each of the outlines above usually becomes a function.
-Those functions depend on others,
-some of which are written to break code into comprehensible chunks and are then called just once,
-others of which are utilities that may be called many times from many different places.
+Don't be sarcastic or disparaging.
+:   "Did you maybe think about *testing* this garbage?"
+    is a Code of Conduct violation in any well-run project.
+
+Don't present opinions as facts.
+:   "Nobody uses X any more" might be true.
+    If it is,
+    the person making the claim ought to be able to point at download statistics
+    or a Google Trends search;
+    if they can't,
+    they should say,
+    "I don't think anybody uses X any more" and explain why they think that.
 
-FIXME: figure
+Don't feign surprise or pass judgment.
+:   "Gosh, didn't you know [some obscure fact]?" isn't helpful;
+    neither is, "Geez, why don't you [some clever trick] here?"
 
-Different people order these differently;
-our preferred order is:
+Don't overwhelm people with details.
+:   If someone has used the letter `x` as a variable name in several places,
+    and they shouldn't have,
+    comment on the first two or three and simply put a check beside the others—the reader
+    won't need the comment repeated.
 
-1.  Put all of the single-use functions in the first half of the file
-    in the order in which they are likely to be called.
-2.  Put all of the multi-use utility functions in the bottom of the file in alphabetical order.
+Don't ask people to do extra work.
+:   Nobody enjoys fixing bugs and style violations.
+    Asking them to add a few features while they're at it is rude.
 
-If any of those utility functions are used by other scripts or programs,
-they should go in a file of their own.
-In fact,
-this is a good practice even if they're only used by one program,
-since it signals even more clearly which functions are in the "structural" layer
-and which are in the "utility" layer.
+Don't let people break these rules just because they're frequent contributors or in positions of power.
+:   The culture of any organization is shaped by the worst behavior it is willing to tolerate @Grue2015.
+    If you let people be rude to one another,
+    *that* is your culture.
+
+Be specific in replies to reviewers.
```

Comment:
Do you think it would be helpful to add an additional heading for how to receive a review? It would include the two points below and maybe also something along the lines of:

> Give people the benefit of the doubt, someone might have left comments that seem short but people (especially those volunteering in open source projects) often have little time to write elaborate replies to everyone's questions and suggestions. Asking for clarification is more productive than becoming defensive, often no harm is meant.

Rephrase to sound better.

Reply:
Done.